service/header/pubsub_manager.go

```diff
@@ -0,0 +1,66 @@
+package header
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+// PubSubTopic hardcodes the name of the ExtendedHeader
+// gossipsub topic.
+const PubSubTopic = "header-sub"
+
+// topicValidator represents a topic validator function for a pubsub topic.
+type topicValidator func(ctx context.Context, p peer.ID, msg *pubsub.Message) pubsub.ValidationResult
+
+// PubsubManager manages the lifecycle and relationship of header Service
+// with the "header-sub" gossipsub topic.
+type PubsubManager struct {
```

Comment:
Call it a P2PSubscriber, as it implements a Subscriber interface and works on the p2p layer.

Reply:
done